### PR TITLE
Feature/millora visualitzacio rols

### PR DESCRIPTION
**Pull request by spuigmitjadxc**
Customize admin views related to department admins, process admins, process admins and assemblies admins.
Customize Process and Assembly views.

This PR branches off from #24 . So #24  should be merged into master before the present PR.

### CODE REVIEW

File: app/decorators/decidim/participatory_process_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+#
+# This decorator adds required associations between Decidim::ParticipatoryProcess and Departaments.
+#
+#
+require_dependency 'decidim/participatory_process'
+Decidim::ParticipatoryProcess.class_eval do
+  belongs_to  :areas,
```
Comment (Leusev) [inline app/decorators/decidim/participatory_process_decorator.rb:9]: Why do we need this belongs_to as we already have area attribute @spuigmitjadxc ?

File: app/decorators/decidim/assemblies_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+#
+# This decorator adds required associations between Decidim::ParticipatoryProcess and Departaments.
+#
+#
+require_dependency 'decidim/assembly'
+Decidim::Assembly.class_eval do
+  belongs_to  :areas,
```
Comment (Leusev) [inline app/decorators/decidim/assemblies_decorator.rb:9]: Why do we need this belongs_to as we already have area attribute @spuigmitjadxc  ?

File: app/decorators/decidim/participatory_process_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+              class_name: 'Areas',
+              foreign_key: 'decidim_department_id'
+
+  belongs_to  :participatory_process_group,
```
Comment (Leusev) [inline app/decorators/decidim/participatory_process_decorator.rb:13]: This field already exists in the original participatory_process class:
- https://github.com/decidim/decidim/blob/c964194a0c8b79374f00bfdef1678c3457c8a523/decidim-participatory_processes/app/models/decidim/participatory_process.rb#L30
so, why do we need to add it again @spuigmitjadxc ?

File: app/decorators/decidim/assemblies_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+              class_name: 'Areas',
```
Comment (Leusev) [inline app/decorators/decidim/assemblies_decorator.rb:10]: There isn't any model called "Areas"

**Comment by spuigmitjadxc** [comment]
Hello @Leusev ,
I have fix the things that you comment me, 
sorry for that bugs, I'm kind of noobie on rails and decidim,
it seems to be that problem that cause the error on creation of assemblies and processes

File: config/initializers/user_roles.rb [hunk 5]
```diff
@@ -1,2 +1,3 @@
 Decidim.user_roles << 'department_admin'
+Decidim.user_roles << 'space_admin'
```
Comment (tramuntanal) [inline config/initializers/user_roles.rb:2]: @spuigmitjadxc what is this new role for?
This PR is supposed to show users with existing roles, not to create new roles.

File: app/decorators/decidim/user_decorator.rb [hunk 6]
```diff
@@ -12,11 +12,28 @@
...
+  has_and_belongs_to_many :assemblies,
+                          join_table: :decidim_assembly_user_roles,
+                          foreign_key: :decidim_user_id,
+                          association_foreign_key: :decidim_assembly_id,
+                          validate: false
+
   scope :admins, -> { where(admin: true) }
   scope :user_managers, -> { where(roles: ["user_manager"]) }
   scope :department_admins, -> { where(roles: ["department_admin"]) }
+  scope :space_administrator, -> { where(roles: ["department_admin"]) }
```
Comment (tramuntanal) [inline app/decorators/decidim/user_decorator.rb:30]: is this correct?

File: app/decorators/decidim/user_decorator.rb [hunk 7]
```diff
@@ -12,11 +12,28 @@
...
 
   def department_admin?
     role?('department_admin')
   end
+
+  def space_administrator?
+    role?('space_administrator')
```
Comment (tramuntanal) [inline app/decorators/decidim/user_decorator.rb:37]: where is that this role is assigned to users?
also `space_administrator` name is different from the one used in other places like:
- app/queries/decidim/admin/user_admin_filter.rbL61
- app/views/decidim/admin/users/_form.html.erb, 
- config/initializers/user_roles.rbL2

File: app/queries/decidim/admin/user_admin_filter.rb [hunk 8]
```diff
@@ -8,41 +8,62 @@ class UserAdminFilter < Rectify::Query
...
+        users.where('(id in (select decidim_user_id ' + 
+                        ' from decidim_participatory_process_user_roles ' +
+                        ' where decidim_participatory_process_id in (select id' +
+                         ' from decidim_participatory_processes' +
+                         ' where lower(title::text) like lower(?)))' +
+                      ' or id in  ( select decidim_user_id' +
+                      ' from decidim_assembly_user_roles' +
+                      ' where decidim_assembly_id in (select id' +
+                        ' from decidim_assemblies' +
+                        ' where lower(title::text) like lower(?))))', "%#{process_name}%", "%#{process_name}%")
```
Comment (tramuntanal) [inline app/queries/decidim/admin/user_admin_filter.rb:56]: we should never interpolate values to queries. Please, add the `%` in the string and simply pass the values as arguments:
```suggestion
        users.where('(id in (select decidim_user_id ' + 
                        ' from decidim_participatory_process_user_roles ' +
                        ' where decidim_participatory_process_id in (select id' +
                         ' from decidim_participatory_processes' +
                         ' where lower(title::text) like lower(%?%)))' +
                      ' or id in  ( select decidim_user_id' +
                      ' from decidim_assembly_user_roles' +
                      ' where decidim_assembly_id in (select id' +
                        ' from decidim_assemblies' +
                        ' where lower(title::text) like lower(%?%))))', process_name, process_name)
```

Comment (tramuntanal) [reply app/queries/decidim/admin/user_admin_filter.rb:56]: also you can use the `ilike` operator and get rid of the `lower` functions.

File: app/decorators/decidim/admin/users_controller_decorator.rb [hunk 9]
```diff
@@ -19,8 +20,76 @@ def user
     @user ||= original_collection.find(params[:id])
   end
 
+  def show
```
Comment (tramuntanal) [inline app/decorators/decidim/admin/users_controller_decorator.rb:23]: I think that the indentation is not correct, making it very difficult to read. Can you check it please?

File: app/decorators/decidim/admin/users_controller_decorator.rb [hunk 10]
```diff
@@ -19,8 +20,76 @@ def user
...
+          type = process.participatory_process_group&.name["ca"]
+        end
+      else 
+        type = t("models.user.fields.process_type", scope: "decidim.admin")
+      end
+
+      process_title = process.title[locale]
+      if process_title == '' then
+        process_title = process.title["ca"]
+      end
```
Comment (tramuntanal) [inline app/decorators/decidim/admin/users_controller_decorator.rb:42]: why this kind of logic?
We're lying to the user making her believe that the process has title in the current locale when it is not true..
Is this a customer request?
The same for the next similar cases

File: app/views/decidim/admin/users/_form.html.erb [hunk 11]
```diff
@@ -6,8 +6,10 @@
   <%= form.email_field :email, label: t(".email") %>
 </div>
 
+<% @roles_for_sale = @form.available_roles_for_select.reject{ |el| el[1] == 'space_admin' } %>
```
Comment (tramuntanal) [inline app/views/decidim/admin/users/_form.html.erb:9]: what does it mean `roles_for_sale` ?

File: app/decorators/decidim/admin/users_controller_decorator.rb (hunk 10 above)
Comment (spuigmitjadxc) [reply app/decorators/decidim/admin/users_controller_decorator.rb:42]: I just followed the current logic, if the title is not able in current language it shows it in catalan (default language), you can check that logic is the same in production

File: app/decorators/decidim/admin/users_controller_decorator.rb (hunk 9 above)
Comment (spuigmitjadxc) [reply app/decorators/decidim/admin/users_controller_decorator.rb:23]: I just check it and it seems fine in my editor

File: app/helpers/decidim/admin/user_roles_helper.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Decidim
```
Comment (tramuntanal) [inline app/helpers/decidim/admin/user_roles_helper.rb:3]:
```suggestion
#  NOTE: This module is being override to take into account when a user is a department_admin.

module Decidim
```

**Pull request merged**